packages/common/src/converter/protobuf-payload-converters.ts

```diff
@@ -1,21 +1,22 @@
-import { PayloadConverterError, ValueError } from '../errors';
-import { errorMessage, hasOwnProperties, hasOwnProperty, isRecord } from '../type-helpers';
 import * as protoJsonSerializer from 'proto3-json-serializer';
 import type { Message, Namespace, Root, Type } from 'protobufjs';
+import { decode, encode } from '../encoding';
+import { PayloadConverterError, ValueError } from '../errors';
+import { Payload } from '../interfaces';
+import { errorMessage, hasOwnProperties, hasOwnProperty, isRecord } from '../type-helpers';
 import {
   BinaryPayloadConverter,
   CompositePayloadConverter,
   JsonPayloadConverter,
   PayloadConverterWithEncoding,
   UndefinedPayloadConverter,
 } from './payload-converter';
-import { Payload } from '../interfaces';
 
-import { EncodingType, encodingTypes, METADATA_ENCODING_KEY, METADATA_MESSAGE_TYPE_KEY, str, u8 } from './types';
+import { encodingTypes, METADATA_ENCODING_KEY, METADATA_MESSAGE_TYPE_KEY } from './types';
 
 abstract class ProtobufPayloadConverter implements PayloadConverterWithEncoding {
   protected readonly root: Root | undefined;
-  public abstract encodingType: EncodingType;
+  public abstract encodingType: string;
```

Comment:
so that users can use their own encoding type?

Reply:
Yes, this type should have never existed, we intentionally don't limit the types of encoding.